lib/cloak/ciphers/aes_ctr.ex

```diff
@@ -86,4 +86,31 @@ defmodule Cloak.Ciphers.AES.CTR do
         false
     end
   end
+
+  # TODO: remove this once support for Erlang/OTP 21 is dropped
+  defp do_init(key, iv, encoder?) do
+    if System.otp_release() >= "22" do
```

Comment:
Hey @v0idpwn, what about checking this at compile time? I think it can suppress some warnings and avoid this conditional. WDYT?

ref: https://github.com/aws-beam/aws-elixir/pull/71/files

Reply:
Thats a good idea. Will push it

Reply:
Fix'd. But we still get a lot of warnings because of the deprecated API, which relies on things that aren't available from OTP 24. I'll address this in another PR.